src/compression.rs

```diff
@@ -1471,20 +1516,28 @@ pub fn compression(
 
     println!("{}","\n=======Compression Summary=======".blue().bold());
     println!("Found {} inventions", step_results.len());
-    println!("Cost Improvement: ({:.2}x better) {} -> {}", compression_factor(programs_expr,&rewritten), programs_expr.cost(), rewritten.cost());
+    println!("Cost Improvement (train): ({:.2}x better) {} -> {}", compression_factor(train_programs_expr, &train_rewritten), train_programs_expr.cost(), train_rewritten.cost());
+    println!("Cost Improvement (test): {}",
+        if let Some(e) = test_programs_expr { format!("{:.2}x better) {} -> {}", compression_factor(e, &test_rewritten.as_ref().unwrap()), e.cost(), test_rewritten.unwrap().cost()) } else { "N/A".to_string() }
+    );
     for i in 0..step_results.len() {
         let res = &step_results[i];
-        println!("{} ({:.2}x wrt orig): {}" ,res.inv.name.clone().blue(), compression_factor(programs_expr, &res.rewritten), res);
+        println!("train: {} ({:.2}x wrt orig): {}", res.inv.name.clone().blue(), compression_factor(train_programs_expr, &res.train_data.rewritten), res);
+        println!("test: {}",
+            if let Some(e) = test_programs_expr { format!("{} ({:.2}x wrt orig): {}", res.inv.name.clone().blue(), compression_factor(e, &res.test_data.as_ref().unwrap().rewritten), res) } else { "N/A".to_string() }
```

Comment:
See comment for line 96 in `compress.rs`. These cases involving "N/A" as the only option used in the `else` branch really highlight that this could be made a lot less verbose with some utility function.

Reply:
Same solution